metadata-ingestion/src/datahub/ingestion/source/sigma/sigma.py

```diff
@@ -263,8 +336,79 @@ def _gen_dashboard_info_workunit(self, page: Page) -> MetadataWorkUnit:
             entityUrn=dashboard_urn, aspect=dashboard_info_cls
         ).as_workunit()
 
+    def _get_element_data_source_platform_details(
+        self, full_path: str
+    ) -> Optional[PlatformDetail]:
+        data_source_platform_details: Optional[PlatformDetail] = None
+        while full_path != "":
+            if full_path in self.config.chart_sources_platform_mapping:
+                data_source_platform_details = (
+                    self.config.chart_sources_platform_mapping[full_path]
+                )
+                break
+            else:
+                full_path = "/".join(full_path.split("/")[:-1])
+        if (
+            not data_source_platform_details
+            and "*" in self.config.chart_sources_platform_mapping
+        ):
+            data_source_platform_details = self.config.chart_sources_platform_mapping[
+                "*"
+            ]
+
+        return data_source_platform_details
+
+    def _get_element_input_details(
+        self, element: Element, workbook: Workbook
+    ) -> Dict[str, Optional[str]]:
+        """
+        Returns dict with keys as the all element input dataset urn and values as their upstream dataset urn
+        """
+        inputs: Dict[str, Optional[str]] = {}
+        sql_parser_in_tables: List[str] = []
+
+        data_source_platform_details = self._get_element_data_source_platform_details(
+            f"{workbook.path}/{workbook.name}/{element.name}"
+        )
+
+        if element.query and data_source_platform_details:
+            try:
+                sql_parser_in_tables = create_lineage_sql_parsed_result(
+                    query=element.query.strip(),
+                    default_db=None,
+                    platform=data_source_platform_details.data_source_platform,
+                    env=data_source_platform_details.env,
+                    platform_instance=data_source_platform_details.platform_instance,
+                ).in_tables
+            except Exception:
+                logging.debug(f"Unable to parse query of element {element.name}")
+
+        # Add sigma dataset as input of element if present
+        # and its matched sql parser in_table as its upsteam dataset
+        for source_id, source_type in element.upstream_sources.items():
+            if source_type == "dataset":
+                source_id = source_id.split("-")[-1]
+                source_dataset_name = self.sigma_api.datasets.get(source_id)
+                if source_dataset_name:
+                    for in_table_urn in sql_parser_in_tables:
+                        if (
+                            DatasetUrn.from_string(in_table_urn).name.split(".")[-1]
+                            in source_dataset_name.lower()
+                        ):
+                            inputs[
+                                self._gen_sigma_dataset_urn(source_id)
+                            ] = in_table_urn
+                            sql_parser_in_tables.remove(in_table_urn)
```

Comment:
we'll probably need to come back to this to generate CLL

Reply:
To generate CLL, we don't have schema metadata for datasets. We do have columns details used by elements/charts, but can't map that columns with there tables.